exporter/googlecloudexporter/config.go

```diff
@@ -74,121 +67,42 @@ func (c *Config) setClientOptions() {
 		opts = append(opts, option.WithCredentialsFile(c.CredentialsFile))
 	}
 
-	c.GCPConfig.GetClientOptions = func() []option.ClientOption {
-		return opts
-	}
+	return opts
 }
 
 // createDefaultConfig creates the default config for the exporter
 func createDefaultConfig() config.Exporter {
-	defaultNamespace, _ := os.Hostname()
 	return &Config{
 		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
-		Location:         defaultLocation,
-		Namespace:        defaultNamespace,
 		GCPConfig:        createDefaultGCPConfig(),
 		BatchConfig:      createDefaultBatchConfig(),
-		NormalizeConfig:  createDefaultNormalizerConfig(),
-		DetectorConfig:   createDefaultDetectorConfig(),
-		AttributerConfig: createDefaultAttributerConfig(),
-		TransposerConfig: createDefaultTransposerConfig(),
+		DetectConfig:     createDefaultDetectConfig(),
 	}
 }
 
 // createDefaultGCPConfig creates a default GCP config
-func createDefaultGCPConfig() *gcp.LegacyConfig {
-	gcpFactory := gcp.NewFactory()
-	gcpConfig := gcpFactory.CreateDefaultConfig().(*gcp.LegacyConfig)
-	gcpConfig.RetrySettings.Enabled = false
-	gcpConfig.UserAgent = defaultUserAgent
-	gcpConfig.MetricConfig.Prefix = defaultMetricPrefix
-	gcpConfig.ResourceMappings = []gcp.ResourceMapping{
-		{
-			TargetType: genericNodeResource,
-			LabelMappings: []gcp.LabelMapping{
-				{
-					SourceKey: "host.name",
-					TargetKey: "node_id",
-				},
-				{
-					SourceKey: "location",
-					TargetKey: "location",
-				},
-				{
-					SourceKey: "namespace",
-					TargetKey: "namespace",
-				},
-			},
-		},
-	}
-
-	return gcpConfig
+func createDefaultGCPConfig() *gcp.Config {
+	factory := gcp.NewFactory()
+	config := factory.CreateDefaultConfig().(*gcp.Config)
+	config.RetrySettings.Enabled = false
+	config.UserAgent = defaultUserAgent
+	config.MetricConfig.Prefix = defaultMetricPrefix
+	config.LogConfig.DefaultLogName, _ = os.Hostname()
+	return config
 }
 
 // createDefaultBatchConfig creates a default batch config
 func createDefaultBatchConfig() *batchprocessor.Config {
-	batchFactory := batchprocessor.NewFactory()
-	batchConfig := batchFactory.CreateDefaultConfig().(*batchprocessor.Config)
-	return batchConfig
-}
-
-// createDefaultNormalizerConfig creates a default normalizer config
-func createDefaultNormalizerConfig() *normalizesumsprocessor.Config {
-	normalizeFactory := normalizesumsprocessor.NewFactory()
-	return normalizeFactory.CreateDefaultConfig().(*normalizesumsprocessor.Config)
+	factory := batchprocessor.NewFactory()
```

Comment:
extremely minor: any reason why you give these variables prefixes in other create config methods but not here?

Reply:
I think it was just stylistic. I can switch it back if you think it's better the other way. My thinking was that it's in a function called createDefaultBatchConfig and it's the only factory being made, so no need to specify what kind of factory it is. I think I should have done it this way the first time around.

Reply:
@jmwilliams89 i don't think one way is better than the other; my comment was more about consistency. i noticed at least one function which had it named the other way.

Reply:
Good catch. I copied that function over from the old implementation and forgot to make that same style change. It should be consistent now.